### PR TITLE
Initial Commit for Qualys web application scanner

### PR DESCRIPTION
**Pull request by paragkhatavkar**

### CODE REVIEW

File: connector.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from connectors.core.connector import Connector, get_logger, ConnectorError
+
+from .operations import operations, _check_health
+
+logger = get_logger('qualyswas')
```
Comment (cs-prerna-joshi) [inline connector.py]:
```suggestion
logger = get_logger('qualys-was')
```

File: info.json [hunk 2]
```diff
@@ -0,0 +1,1227 @@
+{
+  "description": "Cloud solution for continuous web app discovery and detection of vulnerabilities and misconfigurations.",
+  "category": "Web Application Scanner",
+  "name": "qualyswas",
```
Comment (cs-prerna-joshi) [inline info.json]:
```suggestion
  "name": "qualys-web-application-scanner",
```

File: info.json [hunk 3]
```diff
@@ -0,0 +1,1227 @@
...
+  "label": "Qualys Web Scan",
```
Comment (cs-prerna-joshi) [inline info.json]:
```suggestion
  "label": "Qualys Web Application Scan",
```

File: info.json [hunk 4]
```diff
@@ -0,0 +1,1227 @@
...
+        "editable": true,
+        "visible": true,
+        "value": true,
+        "description": "Specifies whether the SSL certificate for the server is to be verified or not."
+      }
+    ]
+  },
+  "operations": [
+    {
+      "title": "Scan Count",
```
Comment (cs-prerna-joshi) [inline info.json]:
```suggestion
      "title": "Get Scan Count",
```

File: info.json [hunk 5]
```diff
@@ -0,0 +1,1227 @@
...
+          "title": "Web App Name",
+          "required": false,
+          "editable": true,
+          "visible": true,
+          "type": "text",
+          "name": "webApp.name",
+          "description": "The name of the web application being scanned."
+        },
+        {
+          "title": "Web App id",
```
Comment (cs-prerna-joshi) [inline info.json]:
```suggestion
          "title": "Web App ID",
```

File: info.json [hunk 6]
```diff
@@ -0,0 +1,1227 @@
...
+          "required": false,
+          "editable": true,
+          "visible": true,
+          "type": "integer",
+          "name": "webApp.id",
+          "description": "The ID of the web application being scanned."
+        },
+        {
+          "title": "Web App Rag ID",
```
Comment (cs-prerna-joshi) [inline info.json]:
```suggestion
          "title": "Web App Tag ID",
```

File: info.json [hunk 7]
```diff
@@ -0,0 +1,1227 @@
...
+          "title": "Web App Name",
+          "required": false,
+          "editable": true,
+          "visible": true,
+          "type": "text",
+          "name": "webApp.name",
+          "description": "The name of the web application being scanned."
+        },
+        {
+          "title": "Web App id",
```
Comment (cs-prerna-joshi) [inline info.json]:
```suggestion
          "title": "Web App ID",
```

File: info.json [hunk 8]
```diff
@@ -0,0 +1,1227 @@
...
+          "required": false,
+          "editable": true,
+          "visible": true,
+          "type": "integer",
+          "name": "webApp.id",
+          "description": "The ID of the web application being scanned."
+        },
+        {
+          "title": "Web App Rag ID",
```
Comment (cs-prerna-joshi) [inline info.json]:
```suggestion
          "title": "Web App Tag ID",
```

File: operations.py [hunk 9]
```diff
@@ -0,0 +1,292 @@
+import base64
+
+import requests
+from connectors.core.connector import get_logger, ConnectorError
+from requests.auth import HTTPBasicAuth
+
+logger = get_logger('qualyswas')
```
Comment (cs-prerna-joshi) [inline operations.py]:
```suggestion
logger = get_logger('qualys-was')
```

File: operations.py [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+            403: 'Access Denied',
+            404: 'The requested resource was not found',
+            409: 'The requested settings conflict with the current settings',
+            410: 'Cannot find the specified object',
+            422: 'Unable to process the request because system lockdown is currently disabled, or some file fingerprint lists or file names were already assigned',
+            423: 'The resource to update is locked and cannot be updated',
+            500: 'Internal Server Error',
+            503: 'Service Unavailable',
+            'time_out': 'The request timed out while trying to connect to the remote server',
+            'ssl_error': 'SSL certificate validation failed'}
```
Comment (cs-prerna-joshi) [inline operations.py]: instead of custom error messages, pass the error message from api.

File: operations.py [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+        self.aouth_cred = {'client_id': '', 'secret_id': '', 'admin_id': '', 'token_expiration': '', 'token': ''}
```
Comment (cs-prerna-joshi) [inline operations.py]: this is not used. you can remove it.

File: operations.py [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
+            if self.error_msg[response.status_code]:
+                logger.error('{}'.format(response.content))
+                raise ConnectorError('{}'.format(self.error_msg[response.status_code]))
+            response.raise_for_status()
+        except requests.exceptions.SSLError as e:
+            logger.exception('{}'.format(e))
+            raise ConnectorError('{}'.format(self.error_msg['ssl_error']))
+        except requests.exceptions.ConnectionError as e:
+            logger.exception('{}'.format(e))
+            raise ConnectorError('{}'.format(self.error_msg['time_out']))
```
Comment (cs-prerna-joshi) [inline operations.py]:
```suggestion
            except req_exceptions.SSLError:
        logger.error('An SSL error occurred')
        raise ConnectorError('An SSL error occurred')
    except req_exceptions.ConnectionError:
        logger.error('A connection error occurred')
        raise ConnectorError('A connection error occurred')
    except req_exceptions.Timeout:
        logger.error('The request timed out')
        raise ConnectorError('The request timed out')
    except req_exceptions.RequestException:
        logger.error('There was an error while handling the request')
        raise ConnectorError('There was an error while handling the request')
```

File: info.json (hunk 2 above)
Comment (paragkhatavkar) [reply info.json]: done

File: connector.py (hunk 1 above)
Comment (paragkhatavkar) [reply connector.py]: done

File: info.json (hunk 3 above)
Comment (paragkhatavkar) [reply info.json]: done

File: info.json (hunk 4 above)
Comment (paragkhatavkar) [reply info.json]: done

File: info.json (hunk 5 above)
Comment (paragkhatavkar) [reply info.json]: done

File: info.json (hunk 6 above)
Comment (paragkhatavkar) [reply info.json]: done

File: info.json (hunk 7 above)
Comment (paragkhatavkar) [reply info.json]: done

File: info.json (hunk 8 above)
Comment (paragkhatavkar) [reply info.json]: done

File: operations.py (hunk 9 above)
Comment (paragkhatavkar) [reply operations.py]: done

File: operations.py (hunk 10 above)
Comment (paragkhatavkar) [reply operations.py]: done

File: operations.py (hunk 11 above)
Comment (paragkhatavkar) [reply operations.py]: done

File: operations.py (hunk 12 above)
Comment (paragkhatavkar) [reply operations.py]: done

**Review by cs-prerna-joshi — APPROVED** [review]
Need to remove the qualyswas

**Pull request merged**